pkg/agent/multicast/mcast_controller_test.go

```diff
@@ -121,26 +127,132 @@ func TestUpdateGroupMemberStatus(t *testing.T) {
 	}
 	mctrl.addGroupMemberStatus(event)
 	mockOFClient.EXPECT().SendIGMPQueryPacketOut(igmpQueryDstMac, types.McastAllHosts, uint32(0), gomock.Any()).Times(len(queryVersions))
-	for _, e := range []*mcastGroupEvent{
-		{group: mgroup, eType: groupJoin, time: event.time.Add(time.Second * 20), iface: if1},
-		{group: mgroup, eType: groupJoin, time: event.time.Add(time.Second * 40), iface: if1},
-		{group: mgroup, eType: groupJoin, time: event.time.Add(time.Second * 60), iface: if2},
-		{group: mgroup, eType: groupLeave, time: event.time.Add(time.Second * 61), iface: if1},
-		{group: mgroup, eType: groupLeave, time: event.time.Add(time.Second * 62), iface: if2},
+	for _, e := range []struct {
+		name       string
+		groupEvent mcastGroupEvent
+	}{
+		{
+			name: "if1 joins group after 20 seconds",
+			groupEvent: mcastGroupEvent{
+				group: mgroup, eType: groupJoin, time: event.time.Add(time.Second * 20), iface: if1,
+			},
+		},
+		{
+			name: "if1 joins group after 40 seconds",
+			groupEvent: mcastGroupEvent{
+				group: mgroup, eType: groupJoin, time: event.time.Add(time.Second * 40), iface: if1,
+			},
+		},
+		{
+			name: "if2 joins group after 60 seconds",
+			groupEvent: mcastGroupEvent{
+				group: mgroup, eType: groupJoin, time: event.time.Add(time.Second * 60), iface: if2,
+			},
+		},
+		{
+			name: "if1 leaves group after 61 seconds",
+			groupEvent: mcastGroupEvent{
+				group: mgroup, eType: groupLeave, time: event.time.Add(time.Second * 61), iface: if1,
+			},
+		},
+		{
+			name: "if2 leaves group after 62 seconds",
```

Comment:
It seems the test cases in `TestUpdateGroupMemberStatus` are not independent: they are a series of events. The "61 seconds case" tests one interface leaves the group, and "62 seconds case" tests the last interface leaves the group.

Any time greater than 60 seconds should work for "62 seconds and 61 seconds cases" because the leave event won't affect `lastIGMPReport.`